packages/semaphore/src/Semaphore.ts

```diff
@@ -0,0 +1,43 @@
+export class Permit {
```

Comment:
I think this would be more readable with one class per file. The tests could be split this way as well

Reply:
I didn't want to elevate Permit to a top-level concept. The only reason I ended up defining this as a class here was to make TypeScript-ifying things easier. In my original implementation this is just a lightweight YOLO object.